File: app/src/main/java/com/example/hostelmanagementsystem/models/Complaint.kt
package com.example.hostelmanagementsystem.models

data class Complaint(
    val id: String,
    val complaintId: String,
    val subject: String,
    val description: String,
    val status: String,
    val adminNotes: String,
    val resolvedAt: String?,
    val createdAt: String?,
    val updatedAt: String?,
    val user: User?
)

File: backend/models/Complaint.js
const mongoose = require("mongoose");

const complaintSchema = new mongoose.Schema(
  {
    complaintId: {
      type: String,
      required: true,
      unique: true
    },
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },
    subject: {
      type: String,
      required: true,
      trim: true
    },
    description: {
      type: String,
      required: true,
      trim: true
    },
    status: {
      type: String,
      enum: ["open", "accepted", "in_progress", "resolved", "closed"],
      default: "open"
    },
    adminNotes: {
      type: String,
      default: "",
      trim: true
    },
    resolvedAt: {
      type: Date
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model("Complaint", complaintSchema);
